libcsl_godot/src/wallet.rs

```diff
@@ -613,62 +629,126 @@ impl SingleAddressWalletStore {
                     description: description.clone(),
-                    public_key: PackedByteArray::from(new_account_key.as_bytes().as_slice()),
+                    network,
+                    address,
+                    address_bech32,
                 };
-                (new_account, new_account_key)
+                Ok((new_account, new_account_key))
             },
         )
     }
+
+    #[func]
+    fn get_accounts(&self) -> Array<Gd<Account>> {
+        self.accounts.values().cloned().collect()
+    }
+
+    // EXPORT WRAPPERS
+
+    #[func]
+    fn _import_from_seedphrase(
+        phrase: String,
+        mnemonic_password: PackedByteArray,
+        wallet_password: PackedByteArray,
+        account_index: u32,
+        account_name: String,
+        account_description: String,
+        network: u8,
+    ) -> Gd<GResult> {
+        Self::to_gresult_class(Self::import_from_seedphrase(
+            phrase,
+            mnemonic_password,
+            wallet_password,
+            account_index,
+            account_name,
+            account_description,
+            network,
+        ))
+    }
+
+    #[func]
+    fn _import_from_resource(resource: Gd<Resource>, network: u8) -> Gd<GResult> {
+        Self::to_gresult_class(Self::import_from_resource(resource, network))
+    }
+
+    #[func]
+    fn _create(
+        wallet_password: PackedByteArray,
+        account_index: u32,
+        account_name: String,
+        account_description: String,
+        network: u8,
+    ) -> Gd<GResult> {
+        Self::to_gresult_class(Self::create(
+            wallet_password,
+            account_index,
+            account_name,
+            account_description,
+            network,
+        ))
+    }
+
+    #[func]
+    pub fn _add_account(
+        &mut self,
+        account: u32,
+        name: GString,
+        description: GString,
+        password: PackedByteArray,
+    ) -> Gd<GResult> {
+        Self::to_gresult_class(self.add_account(account, name, description, password))
+    }
+
+    #[func]
+    pub fn _get_wallet(&self, account_index: u32) -> Gd<GResult> {
+        Self::to_gresult_class(self.get_wallet(account_index))
+    }
 }
 
-/// An account, as stored within a `SingleAddressWalletStore`. This is a data
-/// class and should not be manipulated directly
+/// An account as tracked inside `SingleAddressWallet`.
 #[derive(GodotClass)]
-#[class(base=Resource)]
+#[class(base=RefCounted, rename=_Account)]
 pub struct Account {
     #[var]
     index: u32,
     #[var]
     name: GString,
     #[var]
     description: GString,
-    #[var]
-    public_key: PackedByteArray,
-}
-
-// Used internally inside `SingleAddressWallet` to track account information.
-#[derive(GodotClass)]
-#[class(base=RefCounted)]
-pub struct AccountInfo {
-    index: u32,
-    name: GString,
-    description: GString,
     public_key: Bip32PublicKey,
     address: CSLAddress,
+    #[var]
     address_bech32: GString,
```

Comment:
Creating a Bech32 string from a `CSLAddress` can fail (and AFAIK this is also the case for our own `Address` struct), so the goal here is to pre-compute the Bech32 string when creating the wallet. In this way, `get_address_bech32` can now return just the Bech32 string instead of a `Result`, making the API a bit nicer.